core/jobargument.go

```diff
@@ -75,56 +66,61 @@ func (a *JobArgument) Append(s string, isS3path bool) *JobArgument {
 	return a
 }
 
-func (a *JobArgument) CheckConditionals(wp *WorkerParams, src *JobArgument, opts opt.OptionList) (ret error) {
+// CheckConditions checks if the job satisfies the conditions if the job has -n, -s and -u flags.
+// It returns error-embedded JobResponse with status "warning" if none of the requirements are met.
+// It returns nil if any warning or error is encountered during this check.
+func (a *JobArgument) CheckConditions(wp *WorkerParams, src *JobArgument, opts opt.OptionList) *JobResponse {
+	var res *JobResponse
+
 	if opts.Has(opt.IfNotExists) {
 		ex, err := a.Exists(wp)
 		if err != nil {
-			return err
+			return &JobResponse{status: statusErr, err: err}
 		}
 		if ex {
-			ret = ErrObjectExistsButOk
+			res = &JobResponse{status: statusWarning, err: ErrObjectExists}
 		} else {
-			ret = nil
+			res = nil
 		}
 	}
 
 	if opts.Has(opt.IfSizeDiffers) {
 		sDest, err := a.Size(wp)
 		if err != nil {
-			return err
+			return &JobResponse{status: statusErr, err: err}
 		}
 
 		sSrc, err := src.Size(wp)
 		if err != nil {
-			return err
+			return &JobResponse{status: statusErr, err: err}
 		}
 
 		if sDest == sSrc {
-			ret = ErrObjectSizesMatchButOk
+			res = &JobResponse{status: statusWarning, err: ErrObjectSizesMatch}
 		} else {
-			ret = nil
+			res = nil
 		}
 	}
 
 	if opts.Has(opt.IfSourceNewer) {
 		tDest, err := a.ModTime(wp)
 		if err != nil {
-			return err
+			return &JobResponse{status: statusErr, err: err}
 		}
 
 		tSrc, err := src.ModTime(wp)
 		if err != nil {
-			return err
+			return &JobResponse{status: statusErr, err: err}
```

Comment:
we can use `jobResponse` helper in this file as well.

Reply:
I used jobResponse() helper on all responses.